packages/agent/app/test-frame.ts

```diff
@@ -1,30 +1,48 @@
 import { Operation } from 'effection';
-import { bigtestGlobals } from '@bigtest/globals';
 import { once } from '@effection/events';
+import { Channel } from '@effection/channel';
+import { subscribe } from '@effection/subscription';
+import { TestEvent, Run } from '../shared/protocol';
+import { setLaneConfigFromAgentFrame } from './lane-config';
 
 export class TestFrame {
-  static *start(): Operation<TestFrame> {
-    let element = document.getElementById('test-frame') as HTMLIFrameElement;
-
-    bigtestGlobals.testFrame = element;
+  static setup(id: string, src: string): TestFrame {
+    let element = document.getElementById(id);
+    if (!element) {
+      throw new Error(`CRITICAL: unabled to find the test-frame! It should be findable with id: ${id}`);
+    }
+    if (!(element instanceof HTMLIFrameElement)) {
+      throw new Error(`CRITICAL: expected the test frame to be an <iframe>, but it was ${element}`);
+    }
 
-    return new TestFrame(element);
+    return new TestFrame(element, src);
   }
```

Comment:
Does this even need to be a class? Can we just have `element` be a getter which throws an error if the frame is absent?